internal/extension/reconciler.go

```diff
@@ -141,6 +146,50 @@ func (d *StateAwareDAG) FindGatewaysFor(targetRefs []*extpb.TargetRef) ([]*extpb
 		return gw.GetMetadata().GetNamespace() + "/" + gw.GetMetadata().GetName()
 	}), nil
 }
+func (d *StateAwareDAG) FindPoliciesFor(targetRefs []*extpb.TargetRef, policyType machinery.Policy) ([]*extpb.Policy, error) {
+	initialTargets := d.topology.All().Items(func(o machinery.Object) bool {
+		return len(lo.Filter(targetRefs, func(t *extpb.TargetRef, _ int) bool {
+			return t.Name == o.GetName() && t.Kind == o.GroupVersionKind().Kind
+		})) > 0
+	})
+
+	chain := make([]machinery.Object, 0)
```

Comment:
I've seen that for the Gateways you do similar things, like building an initial chain, then finding/gathering matching objects... in a future refactor we might want to have these procedures in a func.